specs/lightclient/beacon-chain.md

```diff
@@ -0,0 +1,186 @@
+  - [Beacon state accessors](#beacon-state-accessors)
+    - [`get_sync_committee_indices`](#get_sync_committee_indices)
+    - [`get_sync_committee`](#get_sync_committee)
+  - [Block processing](#block-processing)
+    - [Sync committee processing](#sync-committee-processing)
+  - [Epoch processing](#epoch-processing)
+    - [Final updates](#final-updates)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is a standalone beacon chain patch adding light client support via sync committees.
+
+## Constants
+
+| Name | Value |
+| - | - | 
+| `BASE_REWARDS_PER_EPOCH` | `uint64(5)` |
+
+## Configuration
+
+### Misc
+
+| Name | Value |
+| - | - | 
+| `SYNC_COMMITTEE_SIZE` | `uint64(2**10)` (= 1024) |
+| `SYNC_COMMITTEE_PUBKEY_AGGREGATES_SIZE` | `uint64(2**6)` (= 64) |
+
+### Time parameters
+
+| Name | Value | Unit | Duration |
+| - | - | :-: | :-: |
+| `EPOCHS_PER_SYNC_COMMITTEE_PERIOD` | `Epoch(2**8)` (= 256) | epochs | ~27 hours |
+
+### Domain types
+
+| Name | Value |
+| - | - |
+| `DOMAIN_SYNC_COMMITTEE` | `DomainType('0x07000000')` |
+
+## Containers
+
+### Extended containers
+
+#### `BeaconBlockBody`
+
+```python
+class BeaconBlockBody(phase0.BeaconBlockBody):
+    # Sync committee aggregate signature
+    sync_committee_bits: Bitvector[SYNC_COMMITTEE_SIZE]
+    sync_committee_signature: BLSSignature
+```
+
+#### `BeaconState`
+
+```python
+class BeaconState(phase0.BeaconState):
+    # Sync committees
+    current_sync_committee: SyncCommittee
+    next_sync_committee: SyncCommittee
+```
+
+### New containers
+
+#### `SyncCommittee`
+
+```python
+class SyncCommittee(Container):
+    pubkeys: Vector[BLSPubkey, SYNC_COMMITTEE_SIZE]
+    pubkey_aggregates: Vector[BLSPubkey, SYNC_COMMITTEE_SIZE // SYNC_COMMITTEE_PUBKEY_AGGREGATES_SIZE]
+```
+
+## Helper functions
+
+### Beacon state accessors
+
+#### `get_sync_committee_indices`
+
+```python
+def get_sync_committee_indices(state: BeaconState, epoch: Epoch) -> Sequence[ValidatorIndex]:
+    """
+    Return the sync committee indices for a given state and epoch.
+    """
+    base_epoch = Epoch((max(epoch // EPOCHS_PER_SYNC_COMMITTEE_PERIOD, 1) - 1) * EPOCHS_PER_SYNC_COMMITTEE_PERIOD)
+    active_validator_indices = get_active_validator_indices(state, base_epoch)
+    active_validator_count = uint64(len(active_validator_indices))
+    seed = get_seed(state, base_epoch, DOMAIN_SYNC_COMMITTEE)
+    i, sync_committee_indices = 0, []
+    while len(sync_committee_indices) < SYNC_COMMITTEE_SIZE:
+        shuffled_index = compute_shuffled_index(uint64(i % active_validator_count), active_validator_count, seed)
+        candidate_index = active_validator_indices[shuffled_index]
+        random_byte = hash(seed + uint_to_bytes(uint64(i // 32)))[i % 32]
+        effective_balance = state.validators[candidate_index].effective_balance
+        if effective_balance * MAX_RANDOM_BYTE >= MAX_EFFECTIVE_BALANCE * random_byte:
```

Comment:
`MAX_RANDOM_BYTE` is not defined globally

It was previous defined within `compute_proposer_index`